src/components/molecules/AnnouncementMessage/AnnouncementMessage.tsx

```diff
@@ -1,51 +1,88 @@
-import classNames from "classnames";
+import React, { useEffect } from "react";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
+import classNames from "classnames";
 
 import { getLinkFromText } from "utils/getLinkFromText";
 
-import { useChatSidebarControls } from "hooks/chatSidebar";
+import { useShowHide } from "hooks/useShowHide";
+
+import { BannerFormData } from "types/banner";
+
+import { LinkButton } from "components/atoms/LinkButton";
 
 import "./AnnouncementMessage.scss";
 
-type AnnouncementMessageProps = {
-  message?: string;
-};
+export interface AnnouncementMessageProps {
+  banner?: BannerFormData;
+  isUser?: boolean;
+}
 
 export const AnnouncementMessage: React.FC<AnnouncementMessageProps> = ({
-  message = "",
+  banner,
+  isUser = false,
 }) => {
-  const [isVisible, setVisibility] = useState<boolean>(false);
-  const { isExpanded } = useChatSidebarControls();
-
-  const hideAnnouncement = useCallback(() => {
-    setVisibility(false);
-  }, []);
+  const {
+    isShown: isShownAnnouncementMessage,
+    show: showAnnouncementMessage,
+    hide: hideAnnouncementMessage,
+  } = useShowHide();
 
   useEffect(() => {
-    if (message) {
-      setVisibility(true);
+    if (banner?.content) {
+      showAnnouncementMessage();
     }
-  }, [message]);
+  }, [banner, showAnnouncementMessage]);
 
-  if (!isVisible || !message) return null;
+  const noop = (e: React.MouseEvent<HTMLElement>) => {
+    e.stopPropagation();
+  };
+
+  const isActiveButton =
+    banner?.buttonDisplayText && banner?.buttonUrl && banner?.isActionButton;
+
+  const containerClassNames = classNames("AnnouncementContainer", {
+    "AnnouncementContainer--centered": banner?.isFullScreen,
+    "AnnouncementContainer--canceled": !isUser,
+  });
+
+  if (!isUser && !banner?.content)
+    return (
+      <div className="AnnouncementMessage">
+        <span className="AnnouncementMessage__default-text">
+          No announcement
+        </span>
+      </div>
+    );
+
+  if (!isShownAnnouncementMessage || !banner?.content) return null;
 
   return (
-    <div
-      className={classNames("announcement-container", {
-        centered: !isExpanded,
-      })}
-    >
-      <div className="announcement-message">{getLinkFromText(message)}</div>
-      <span className="close-button" onClick={hideAnnouncement}>
-        <FontAwesomeIcon icon={faTimesCircle} />
-      </span>
+    <div className={containerClassNames} onClick={hideAnnouncementMessage}>
+      <div className="AnnouncementMessage" onClick={noop}>
+        {banner?.title && (
+          <h2 className="AnnouncementMessage__title">{banner.title}</h2>
+        )}
+        <div className="AnnouncementMessage__content">
+          {getLinkFromText(banner.content)}
+        </div>
+        {isActiveButton && (
+          <LinkButton
+            href={banner.buttonUrl}
```

Comment:
If I understand correctly this is the reason to make `href` optional. Wouldn't it be better to just check if banner.buttonUrl exists 2 lines above and not show LinkButton if it's not?

Reply:
https://github.com/sparkletown/sparkle/pull/1440/commits/3e1865646713934612d911551296b8a0e951fb26